main.py

```diff
@@ -5,20 +5,23 @@
 from logger import LOGGER
 from state import State
 from action import Action
-from opening import opening
-from stage import stage
+from opening import Opening
+from stage import Stage
 
 def should_continue(state: State) -> bool:
     return state != State.EXIT
 
 def next_action(state: State) -> Action:
-    if state == State.OPENING:
-        return opening(window)
-    if state == State.STAGE:
-        return stage()
-    LOGGER.error(f"Not implemented state: {state}")
-    LOGGER.error("Quit directly")
-    return Action.QUIT
+    match state:
+        case State.OPENING:
+            scene = Opening(name='Opening', window=window)
+        case State.STAGE:
+            scene = Stage(name='Stage', window=window)
```

Comment:
盡量讓每個case的結局都return或都不return
```suggestion
            return Opening(name='Opening', window=window)
        case State.STAGE:
            return Stage(name='Stage', window=window)
```